tools/openmldb_ops.py

```diff
@@ -0,0 +1,350 @@
+                return Status(-1, "recover partition failed")
+        if not executor.RecoverTablePartition(db, table_name, pid, endpoint).OK():
+            log.error(f"recover table partition failed. db {db} name {table_name} pid {pid} endpoint {endpoint}")
+            return Status(-1, "recover table partition failed")
+
+def RecoverTable(executor : Executor, db, table_name) -> Status:
+    if CheckTable(executor, db, table_name).OK():
+        log.info(f"{table_name} in {db} is healthy")
+        return Status()
+    log.info(f"recover {table_name} in {db}")
+    status, table_info = executor.GetTableInfo(db, table_name)
+    if not status.OK():
+        log.warn(f"get table info failed. msg is {status.GetMsg()}")
+        return Status(-1, f"get table info failed. msg is {status.GetMsg()}")
+    partition_dict = executor.ParseTableInfo(table_info)
+    endpoints = set()
+    for record in table_info:
+        endpoints.add(record[3])
+    endpoint_status = {}
+    for endpoint in endpoints:
+        status, result = executor.GetTableStatus(endpoint)
+        if not status.OK():
+            log.warn(f"get table status failed. msg is {status.GetMsg()}")
+            return Status(-1, f"get table status failed. msg is {status.GetMsg()}")
+        endpoint_status[endpoint] = result
+    tid = int(table_info[-1][1])
+    max_pid = int(table_info[-1][2])
+    for pid in range(max_pid + 1):
+        RecoverPartition(executor, db, partition_dict[str(pid)], endpoint_status)
+    # wait op
+    while True:
+        status, result = executor.ShowOpStatus(db, table_name)
+        is_finish = True
+        if status.OK():
+            for record in result:
+                if record[4] == 'kDoing':
+                    value = " ".join(record)
+                    log.info(value)
+                    is_finish = False
+                    break
+        if not is_finish:
+            log.info(f"waiting task")
+            time.sleep(2)
+        else:
+            break
+    status = CheckTable(executor, db, table_name)
+    if status.OK():
+        log.info(f"{table_name} in {db} recover success")
+    else:
+        log.warn(status.GetMsg())
+    return status
+
+def RecoverData(executor : Executor):
+    status, dbs = executor.GetAllDatabase()
+    if not status.OK():
+        log.error("get database failed")
+        return
+    alldb = list(INTERNAL_DB)
+    alldb.extend(dbs)
+    for db in alldb:
+        status, tables = executor.GetAllTable(db)
+        if not status.OK():
+            log.error(f"get all table failed")
+            return
+        for name in tables:
+            if not RecoverTable(executor, db, name).OK():
+                return
+
+def MigratePartition(db : str, partition : Partition, src_endpoint : str, desc_endpoint : str, one_replica : bool) -> Status:
+    if partition.IsLeader():
+        if one_replica and not executor.AddReplica(db, partition.GetName(), partition.GetPid(), desc_endpoint).OK():
```

Comment:
if we add replica in the `desc_endpoint`, we cannot migrate it to `desc_endpoint`, as we don't allow replicas of the same partition co-existing in the same tablet?